src/protocols/bdx/BdxMessages.h

```diff
@@ -158,6 +167,12 @@ struct TransferInit : public BdxMessage
     CHIP_ERROR Parse(System::PacketBufferHandle aBuffer) override;
     Encoding::LittleEndian::BufferWriter & WriteToBuffer(Encoding::LittleEndian::BufferWriter & aBuffer) const override;
     size_t MessageSize() const override;
+    void LogMessage() const override;
+
+private:
+    // These are only needed for logging purpose
```

Comment:
This feels off  to me - this becomes a struct with private members and the default constructor leaves MessageType uninitialized (so I suppose potentially buggy code)

Reply:
`mRangeCtlFlags` can be a public member, with no loss.

mMessageType is an implementation detail. Suggest having `LogMessage(bdx::MessageType) ` method that avoids needing to store this "self-referential" dynamic type info.